Track Buddy/MotionManager.swift

```diff
@@ -65,13 +68,60 @@ class MotionManager: ObservableObject {
     // Store recent rate-limited points to render path representing past G force values
     private var recentPoints: Deque<CGPoint> = [] // TODO: dluo- think about thread safety?
     
-    func pointPath(atScale factor: CGFloat) -> CGMutablePath {
-        let points = recentPoints.map { CGPoint(x: $0.x * factor, y: $0.y * factor) }
+    func pointPath(at scaleFactor: CGFloat) -> CGMutablePath {
+        
+        /* Interpolate points to fill in gaps between rate-limited points to create a smooth tracer line.
+         
+         Using this interpolate-to-smooth approach because generating the line with the raw accelerometer data at 100 Hz
+         can produce visible divergence between the movement of the dot in the graph and the line that is traced under it.
+         
+         */
+        
+        let interpolatedPoints = interpolate(recentPoints, at: scaleFactor)
         let path = CGMutablePath()
-        path.addLines(between: points)
+        path.addLines(between: interpolatedPoints)
         return path
     }
     
+    private func interpolate(_ points: Deque<CGPoint>, at scaleFactor: CGFloat) -> [CGPoint] {
+        guard points.count > 0 else { return [] }
+        let xPoints = points.map { Double($0.x * scaleFactor) }
+        let yPoints = points.map { Double($0.y * scaleFactor) }
+        let numberOfInterpolatedPoints = Int(Double(1) / Parameters.deviceMotionUpdateInterval * Double(points.count) * graphUpdateInterval)
```

Comment:
I believe the `Int` cast will always truncate instead of round? Is that acceptable here?
e.g. 12.9 will truncate to 12 instead of rounding to 13

Reply:
Looking at the Accelerate docs, it appears to be acceptable just to truncate. However I will admit I just blindly casted it as `Int` without thinking about it and will need to remember do so in the future (duh).